### PR TITLE
Respect exclude & includeSourceIds

### PR DESCRIPTION
**Pull request by mutaphysis**
- Added excludeSourceIds
 - Use exclude & include source ids as Array<string>
 - Respect exclude & include source ids from client config
 - Ensure exclude & include are correctly preserved only when
   the are different than the config & non empty

Test with eg. http://localhost:3000/search?disableWheelmapSource=true&excludeSourceIds=QxyZ7vAMS695Bhvii

### CODE REVIEW

**Comment by dakeyras7** [comment]
Other than my comment s'all 👍🏼

File: src/App.js [hunk 1]
```diff
@@ -445,8 +450,18 @@ class App extends React.Component<Props, State> {
     }
 
     // ensure to keep widget/custom whitelabel parameters
-    if (includeSourceIds) {
-      params.includeSourceIds = includeSourceIds;
+    if (includeSourceIds && includeSourceIds.length > 0) {
+      const includeSourceIdsAsString = includeSourceIds.join(',');
+      if (includeSourceIdsAsString !== clientSideConfiguration.includeSourceIds.join(',')) {
+        params.excludeSourceIds = includeSourceIdsAsString;
```
[the remark this answers is not in the document]
Comment (mutaphysis) [reply src/App.js]: Great catch! Will fix!

[the remark this answers is not in the document]
Comment (mutaphysis) [reply src/App.js]: Fixed

**Comment by mutaphysis** [comment]
@dakeyras7 can you take another look, please?

**Pull request merged**